### PR TITLE
[NETAPI32] Fix SamCloseHandle() and SamDelete*() callers

### PR DESCRIPTION
**Pull request by SergeGautherie**
These functions invalidate the handle.

### CODE REVIEW

**Comment by SergeGautherie** [comment]
Cc @EricKohl

File: dll/win32/netapi32/group_new.c [hunk 1]
```diff
@@ -379,9 +379,11 @@ NetGroupAdd(
     if (ApiStatus == NERR_Success)
     {
         ERR("OpenGroupByName: Group %wZ already exists!\n", &GroupName);
+        ApiStatus = ERROR_GROUP_EXISTS;
 
         SamCloseHandle(GroupHandle);
-        ApiStatus = ERROR_GROUP_EXISTS;
+        GroupHandle = NULL;
```
Comment (GeoB99) [inline dll/win32/netapi32/group_new.c:385]: `SamCloseHandle` already invalidates the handle object for you.
See https://git.reactos.org/?p=reactos.git;a=blob;f=dll/win32/samsrv/samrpc.c;hb=9f3ed60ea4779a8848c3a78d004f415520669832#l160

File: dll/win32/netapi32/group_new.c [hunk 2]
```diff
@@ -646,6 +640,9 @@ NetGroupDel(
         goto done;
     }
 
+    /* A successful delete invalidates the handle */
+    GroupHandle = NULL;
+
```
Comment (GeoB99) [inline dll/win32/netapi32/group_new.c:645]: Now the code below is dead code because you manually invalidated `GroupHandle` (and wrongly so).

**Comment by SergeGautherie** [comment]
Rebased, after hijack 37f38a80de200e2308487993f6d3de25e333e88c.

File: dll/win32/netapi32/group_new.c (hunk 1 above)
Comment (SergeGautherie) [reply dll/win32/netapi32/group_new.c:385]: I am not sure what your point is. You linked to `SamrCloseHandle()`, which is not the same function...

File: dll/win32/netapi32/group_new.c (hunk 2 above)
Comment (SergeGautherie) [reply dll/win32/netapi32/group_new.c:645]: How is my change wrong? How does it create dead code?

File: dll/win32/netapi32/group_new.c [hunk 3]
```diff
@@ -876,7 +873,7 @@ NetGroupEnum(
             TRACE("SamEnumerateGroupsInDomain returned (Status %08lx)\n", Status);
             if (!NT_SUCCESS(Status))
             {
-                ERR("SamEnumerateAliasesInDomain failed (Status %08lx)\n", Status);
+                ERR("SamEnumerateGroupsInDomain() failed. Status 0x%lX\n", Status);
```
Comment (HBelusca) [inline dll/win32/netapi32/group_new.c:876]: Same as above.
```suggestion
                ERR("SamEnumerateGroupsInDomain failed (Status %08lx)\n", Status);
```

File: dll/win32/netapi32/local_group.c [hunk 4]
```diff
@@ -770,6 +771,9 @@ NetLocalGroupDel(
         goto done;
     }
 
+    /* A successful delete invalidates the handle */
+    AliasHandle = NULL;
+
```
Comment (HBelusca) [inline dll/win32/netapi32/local_group.c:776]: Suspiciously wrong, due to the reference counts on the object.
If you want to convince us, prove with a debug trace that when closing the handle on this deleted object, the object becomes instantaneous invalid, whatever the number of handles opened on it...
```suggestion
```

File: dll/win32/netapi32/group_new.c (hunk 2 above)
Comment (HBelusca) [inline dll/win32/netapi32/group_new.c:645]: Suspiciously wrong, due to the reference counts on the object.
If you want to convince us, prove with a debug trace that when closing the handle on this deleted object, the object becomes instantaneous invalid, whatever the number of handles opened on it...
```suggestion
```

File: dll/win32/netapi32/group_new.c [hunk 5]
```diff
@@ -379,9 +379,11 @@ NetGroupAdd(
     if (ApiStatus == NERR_Success)
     {
         ERR("OpenGroupByName: Group %wZ already exists!\n", &GroupName);
+        ApiStatus = ERROR_GROUP_EXISTS;
 
         SamCloseHandle(GroupHandle);
-        ApiStatus = ERROR_GROUP_EXISTS;
+        GroupHandle = NULL;
+
```
Comment (HBelusca) [inline dll/win32/netapi32/group_new.c:386]:
```suggestion
```

File: dll/win32/netapi32/local_group.c [hunk 6]
```diff
@@ -409,9 +409,11 @@ NetLocalGroupAdd(
     if (ApiStatus == NERR_Success)
     {
         ERR("OpenAliasByName: alias %wZ already exists!\n", &AliasName);
+        ApiStatus = ERROR_ALIAS_EXISTS;
 
         SamCloseHandle(AliasHandle);
-        ApiStatus = ERROR_ALIAS_EXISTS;
+        AliasHandle = NULL;
+
```
Comment (HBelusca) [inline dll/win32/netapi32/local_group.c:416]:
```suggestion
```

**Review by HBelusca — CHANGES_REQUESTED** [review]

File: dll/win32/netapi32/group_new.c (hunk 1 above)
Comment (GeoB99) [reply dll/win32/netapi32/group_new.c:385]: https://git.reactos.org/?p=reactos.git;a=blob;f=dll/win32/samlib/samlib.c;hb=6d441f7d65f57c0f5b6da59ae851640a6e787875#l495

Clearly do I have to babysit you on grep'ing stuff now, huh? :trollface:

**Comment by GeoB99** [comment]
Stale PR, so I'm closing this.

**Pull request closed without merging**